File: web/src/interfaces/TransacaoDtos/CreateTransacaoDto.ts
import { TipoTransacao } from "../../enums/TipoTransacaoEnum";
import { empty_Guid } from "../../utils/guid";

export class CreateTransacaoDto {
  id: string = empty_Guid;
  idCategoria?: string;
  idPessoa?: string;
  tipo: TipoTransacao;
  descricao: string;
  receitas: number;
  despesas: number;
  saldo: number;

  constructor(
    tipo: TipoTransacao = TipoTransacao.Receita,
    descricao: string = '',
    receitas: number = 0.0,
    despesas: number = 0.0,
    saldo: number = 0.0,
    idCategoria?: string,
    idPessoa?: string
  ) {
    this.tipo = tipo;
    this.descricao = descricao;
    this.receitas = receitas;
    this.despesas = despesas;
    this.saldo = saldo;
    this.idCategoria = idCategoria;
    this.idPessoa = idPessoa;
  }
}

File: api/HomeBudget.API/HomeBudget.Application/DTOs/TransacaoDtos/CreateTransacaoDto.cs
using HomeBudget.Domain.Entities;
using HomeBudget.Domain.Enums;

namespace HomeBudget.Application.DTOs.TransacaoDtos
{
    public class CreateTransacaoDto
    {
        public Guid Id { get; set; }
        public Guid? IdCategoria { get; set; }
        public Guid? IdPessoa { get; set; }

        public TipoTransacao Tipo { get; set; }

        public string Descricao { get; set; } = string.Empty;
        public double Receitas { get; set; } = 0.00;
        public double Despesas { get; set; } = 0.00;
        public double Saldo { get; set; } = 0.00;
    }
}
